### PR TITLE
introduce casng package

### PR DESCRIPTION
**Pull request by mrahs**
This PR includes one method and some of the scaffolding for other methods that will be posted later.

### CODE REVIEW

File: go/pkg/casng/uploader.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+func MakeWriteResourceName(instanceName, hash string, size int64) string {
+	return fmt.Sprintf("%s/uploads/%s/blobs/%s/%d", instanceName, uuid.New(), hash, size)
+}
+
+// MakeCompressedWriteResourceName returns a valid resource name for writing a compressed blob.
+func MakeCompressedWriteResourceName(instanceName, hash string, size int64) string {
+	return fmt.Sprintf("%s/uploads/%s/compressed-blobs/zstd/%s/%d", instanceName, uuid.New(), hash, size)
+}
+
+// batchingUplodaer implements the corresponding interface.
```
Comment (bozydarsz) [inline go/pkg/casng/uploader.go:53]: nit: should start with capital to match the type name

File: go/pkg/casng/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,54 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "casng",
```
Comment (bozydarsz) [inline go/pkg/casng/BUILD.bazel:4]: what will it do? / what casng stands for?

File: go/pkg/casng/uploader.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+type BatchingUploader struct {
+	*uploader
+}
+
+// streamingUploader implements the corresponding interface.
```
Comment (bozydarsz) [inline go/pkg/casng/uploader.go:58]: ditto

**Review by bozydarsz — APPROVED** [review]
left minor comments, otherwise lgtm

File: go/pkg/casng/BUILD.bazel (hunk 2 above)
Comment (mrahs) [reply go/pkg/casng/BUILD.bazel:4]: This package will eventually include a full upload and download implementations for the CAS API. I wanted to reuse the name `cas`, but that will cause headaches with conflicts with the existing code in that package. The best short name I could come up with is `casng` for CAS next-gen 🤷

**Pull request merged**